src/protego.py

```diff
@@ -0,0 +1,140 @@
+from collections import namedtuple
+from six.moves.urllib.parse import urlparse
+
+Record = namedtuple('Record', ['field', 'value'])
+
+
+class RecordGroup:
+
+    def __init__(self):
+        self.user_agents = []
+        self.rules = []
+        self.crawl_delay = None
+        self.sorted_rules = None
+
+    def add_user_agent(self, user_agent):
+        self.user_agents.append(user_agent.strip().lower())
+
+    def applies_to(self, robotname):
```

Comment:
I’m not sure this logic is correct, at least not according to Google’s specification: https://developers.google.com/search/reference/robots_txt#order-of-precedence-for-user-agents

See this commit for a test function that aims to prove that Protego matches Google’s specification on user-agent string precedence rules: https://github.com/Gallaecio/protego/commit/374219285a434d91ac8f31e118b4ff8f3f030a4e

Assuming Protego aims to abide by (among others) Google’s specification, maybe you could review https://developers.google.com/search/reference/robots_txt and extend Protego’s test suite with tests that cover their specification.